tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -701,6 +803,110 @@ Status StorageManager::object_move(
   return vfs_->move_dir(old_uri, new_uri);
 }
 
+Status StorageManager::get_fragment_info(
+    const ArraySchema* array_schema,
+    uint64_t timestamp,
+    const EncryptionKey& encryption_key,
+    std::vector<FragmentInfo>* fragment_info) {
+  fragment_info->clear();
+
+  // Get fragment URIs
+  std::vector<URI> fragment_uris;
+  RETURN_NOT_OK(get_fragment_uris(array_schema->array_uri(), &fragment_uris));
+
+  // Check if the array is empty
+  if (fragment_uris.empty())
+    return Status::Ok();
+
+  // Sort the URIs by timestamp
+  std::vector<std::pair<uint64_t, URI>> sorted_fragment_uris;
+  get_sorted_fragment_uris(fragment_uris, timestamp, &sorted_fragment_uris);
+
+  // Get the rest of fragment info
+  uint64_t domain_size = 2 * array_schema->coords_size();
+  uint64_t size;
+  bool sparse, in_cache;
+  for (const auto& uri : sorted_fragment_uris) {
+    // Determine if the fragment is sparse
+    URI coords_uri =
+        uri.second.join_path(constants::coords + constants::file_suffix);
+    RETURN_NOT_OK(vfs_->is_file(coords_uri, &sparse));
+
+    // Get fragment size
+    RETURN_NOT_OK(vfs_->dir_size(uri.second, &size));
+
+    // Get fragment non-empty domain
+    auto metadata =
+        new FragmentMetadata(array_schema, !sparse, uri.second, uri.first);
+    RETURN_NOT_OK_ELSE(
+        load_fragment_metadata(metadata, encryption_key, &in_cache),
+        delete metadata);
+    void* non_empty_domain = std::malloc(domain_size);
+    if (non_empty_domain == nullptr) {
+      delete metadata;
+      return LOG_STATUS(Status::StorageManagerError(
+          "Cannot get fragment info; Memory allocation failed"));
+    }
+    std::memcpy(non_empty_domain, metadata->non_empty_domain(), domain_size);
+    delete metadata;
+
+    // Push new fragment info
+    fragment_info->emplace_back(
+        uri.second, sparse, uri.first, size, non_empty_domain, domain_size);
+  }
+
+  return Status::Ok();
+}
+
+Status StorageManager::get_fragment_info(
```

Comment:
No, I don't think so, as this depends on format agnostic parameters. However, retrieving the timestamp from the fragment URI will have to be refactored when we swap the UUID and timestamp in a separate issue.

Reply:
the format defines the binary layout + on disk organization, the two are coupled

Reply:
Still I am not sure I understand why we would have to handle the format version here. We are not doing any parsing of the binary files in this function.

Reply:
Well the filename example is something that will cause a format number increment (so that we can keep reading old arrays). So @jakebolewski might be correct that some checks should be added here, even though the binary format is not touched.

Reply:
Regarding the fragment folder name format, this will be revisited when we swap the UUID with the timestamp. So we will put the necessary checks then. Right now, there is no violation of backwards compatibility.